### PR TITLE
rust-analyzer: update to 2022.06.06

### PR DESCRIPTION
**Pull request by jawadcode**
#### Testing the changes
- I tested the changes in this PR: **YES**|**briefly**|**NO**
YES

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**|**NO**
YES

### CODE REVIEW

File: srcpkgs/rust-analyzer/template [hunk 1]
```diff
@@ -11,7 +11,7 @@ maintainer="Gabriel Sanches <gabriel@gsr.dev>"
 license="Apache-2.0, MIT"
 homepage="https://rust-analyzer.github.io/"
 distfiles="https://github.com/${pkgname}/${pkgname}/archive/${_ver}.tar.gz"
```
Comment (classabbyamp) [inline srcpkgs/rust-analyzer/template:13]:
```suggestion
distfiles="https://github.com/rust-lang/rust-analyzer/archive/${_ver}.tar.gz"
```
r-a is part of the rust namespace now

**Comment by classabbyamp** [comment]
> error: package `xshell-macros v0.2.1` cannot be built because it requires rustc 1.59 or newer, while the currently active rustc version is 1.57.0

did you actually test this? it will fail to build until #37232 is merged.

**Comment by jawadcode** [comment]
> > I tested the changes in this PR: YES|briefly|NO
> YES
> 
> > error: package `xshell-macros v0.2.1` cannot be built because it requires rustc 1.59 or newer, while the currently active rustc version is 1.57.0
> 
> did you actually test this? it will fail to build until #37232 is merged.

Apologies, I misunderstood the question and tested by compiling that version of rust-analyzer from source manually instead of using the xbps tools.

Sorry for wasting your time

**Comment by classabbyamp** [comment]
it's fine, it can still get merged but it will have to wait on the rust update

**Comment by jawadcode** [comment]
> it's fine, it can still get merged but it will have to wait on the rust update

Yeah I'll just leave it open in the meanwhile

**Comment by tranzystorek-io** [comment]
Rust 1.61 is there, I think this PR is good to go ☺️

**Comment by jawadcode** [comment]
> it's fine, it can still get merged but it will have to wait on the rust update

@classabbyamp Would it be possible to merge this PR now since #37232 has been merged?

**Comment by classabbyamp** [comment]
it should build now, yes, but there's still some work to be done with this PR before it can be merged (and I can't merge it anyways, I'm not a maintainer). The things I can see are:

- the comment I made earlier about the distfiles url
- one commit per package (you can use `git squash` and `git push --force`) to fix this, (and in the future merge from master by using `git pull --rebase upstream master` to avoid the merge commit)
- testing what xbps-src builds

**Comment by classabbyamp** [comment]
It seems that f26f39640fcc75516e4e37cb1f4d47f14f27d4e0 undid the version and checksum changes. Can you fix that and squash (as mentioned in my previous comment).

**Comment by classabbyamp** [comment]
superceded by #37566

**Pull request closed without merging**